### PR TITLE
Regenerate go.mod indirect dependencies

### PR DESCRIPTION
**Pull request by gibizer**
During the golang version bump we unintentionally introduced a too new
indirect dependency on kube-openapi:

k8s.io/kube-openapi@v0.0.0-20240209001042-7a0d5b415232

This is not required by any of our direct deps and this is also a
version that required golang 1.21. If your system has golang 1.21
installed the this dependency might force go mod tidy to change the
golang version in the go.mod file to 1.21.

This patch removed all the indirect dependencies and let go mod tidy to
regenerate them to get rid of the too new kube-openapi version.

This happened again in the bump dc2ba9ba and revert 7a0ed7bb4c of the
cert-manager dependency.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -1,3 +1,4 @@
+github.com/benbjohnson/clock v1.1.0/go.mod h1:J11/hYXuz8f4ySSvYwY0FKfm+ezbsZBKZxNJlLklBHA=
```
Comment (jpodivin) [inline go.sum:1]: This library is no longer maintained.

**Review by jpodivin — CHANGES_REQUESTED** [review]
If possible we shouldn't introduce dependency on an unmaintained library, even if it's an indirect one.  

Bumping  zap to version 1.27 should get rid of it. As zap maintainers removed this dependency themselves over a year ago. https://github.com/uber-go/zap/issues/1331

**Comment by gibizer** [comment]
> If possible we shouldn't introduce dependency on an unmaintained library, even if it's an indirect one.
> 
> Bumping zap to version 1.27 should get rid of it. As zap maintainers removed this dependency themselves over a year ago. [uber-go/zap#1331](https://github.com/uber-go/zap/issues/1331)

We are already on zap 1.27 as per https://github.com/openstack-k8s-operators/dataplane-operator/pull/724/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R69 . But some of our deps are using older zap. Hence the go.sum line:

```
❯ go mod graph | grep ' go.uber.org/zap'
github.com/openstack-k8s-operators/dataplane-operator go.uber.org/zap@v1.27.0
github.com/cert-manager/cert-manager@v1.13.4 go.uber.org/zap@v1.25.0
github.com/go-logr/zapr@v1.2.4 go.uber.org/zap@v1.24.0
github.com/openstack-k8s-operators/infra-operator/apis@v0.3.1-0.20240221083751-49edc0df8a12 go.uber.org/zap@v1.26.0
github.com/openstack-k8s-operators/lib-common/modules/certmanager@v0.0.0-20240229121803-169ced56d56e go.uber.org/zap@v1.27.0
github.com/openstack-k8s-operators/lib-common/modules/common@v0.3.1-0.20240229121803-169ced56d56e go.uber.org/zap@v1.27.0
github.com/openstack-k8s-operators/openstack-ansibleee-operator/api@v0.3.1-0.20240226163515-d53e4f43e3b0 go.uber.org/zap@v1.27.0
k8s.io/apiextensions-apiserver@v0.28.7 go.uber.org/zap@v1.19.0
k8s.io/component-base@v0.28.7 go.uber.org/zap@v1.19.0
sigs.k8s.io/controller-runtime@v0.16.5 go.uber.org/zap@v1.25.0
```

**Review by fao89 — COMMENTED** [review]
+2

**Review by rebtoor — APPROVED** [review]

**Review by bshephar — APPROVED** [review]

**Review by fao89 — APPROVED** [review]

**Comment by gibizer** [comment]
and rebased again

**Review by fao89 — APPROVED** [review]

**Review by slagle — APPROVED** [review]

**Pull request merged**